homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,235 @@
+"""Cover entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from motionblindsble.const import MotionBlindType, MotionRunningType
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN, ICON_VERTICAL_BLIND
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class MotionCoverEntityDescription(CoverEntityDescription):
+    """Entity description of a cover entity with default values."""
+
+    key: str = CoverDeviceClass.BLIND.value
+    translation_key: str = CoverDeviceClass.BLIND.value
+
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.HONEYCOMB.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.SHADE
+    ),
+    MotionBlindType.ROMAN.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.SHADE
+    ),
+    MotionBlindType.ROLLER.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.SHADE
+    ),
+    MotionBlindType.DOUBLE_ROLLER.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.SHADE
+    ),
+    MotionBlindType.VENETIAN.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.VENETIAN_TILT_ONLY.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.CURTAIN.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN
+    ),
+    MotionBlindType.VERTICAL.name: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up blind based on a config entry."""
+
+    blind_class = BLIND_TYPE_TO_CLASS[entry.data[CONF_BLIND_TYPE].upper()]
+    device = hass.data[DOMAIN][entry.entry_id]
+    entity = blind_class(device, entry)
+
+    async_add_entities([entity])
+
+
+class GenericBlind(MotionblindsBLEEntity, CoverEntity):
```

Comment:
Would be nice to name the class something to do with the integration since it's hard to figure out which integration is coming from when debugging